c7n/resources/asg.py

```diff
@@ -526,12 +526,20 @@ def get_snapshots(self, ec2, snap_ids):
             try:
                 result = ec2.describe_snapshots(SnapshotIds=snap_ids)
             except ClientError as e:
-                if e.response['Error']['Code'] == 'InvalidSnapshot.NotFound':
-                    msg = e.response['Error']['Message']
-                    e_snap_id = msg[msg.find("'") + 1:msg.rfind("'")]
-                    self.log.warning("Snapshot not found %s" % e_snap_id)
-                    snap_ids.remove(e_snap_id)
-                    continue
+                if e.response['Error']['Code'] in (
+                        'InvalidSnapshot.NotFound', 'InvalidSnapshotID.Malformed'):
+                    if e.response['Error']['Code'] == 'InvalidSnapshotID.Malformed':
+                        msg = e.response['Error']['Message']
```

Comment:
The blocks of code are the same here with the only difference being the warning message.  There is no reason to have an if/else with this much duplicate code.  Please refactor.